dev/Repeater/InspectingDataSource.cpp

```diff
@@ -22,31 +22,41 @@ InspectingDataSource::InspectingDataSource(const winrt::IInspectable& source)
     }
     else
     {
-        // The bindable interop interface are abi compatible with the corresponding
-        // WinRT interfaces.
-        auto bindableVector = source.try_as<winrt::IBindableVector>();
-        if (bindableVector)
+        const auto vectorView = source.try_as<winrt::IVectorView<winrt::IInspectable>>();
```

Comment:
>const auto vectorView = source.try_as<winrt::IVectorView<winrt::IInspectable>>(); [](start = 8, length = 81)

This definition can be made in the if statement right?

Reply:
I think all of the definitions in that area of the code should be refactored (see addional questions part of PR descrition), but I was not sure if there was a reason we did not do that here.

Reply:
ah, I think its just because this code is quite old, I think when it was written we weren't using a c++ compiler that supported definitions in if statements. As for the nested if else statements, I suspect that was just the style of the time, it seems worse to me

Reply:
Would it be fine if we update that in this PR then?

Reply:
I think that would be fine.

Reply:
I agree, refactoring the code to make it more readable is goodness.

Reply:
Updated the code now, it's a lot more readable now and uses const where possible.